griptape/tools/google_docs/tool.py

```diff
@@ -0,0 +1,302 @@
+
+    @activity(
+        config={
+            "description": "Can be used to prepend text to a Google Doc",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_path",
+                        description="Destination file path in the POSIX format. For example, 'foo/bar/baz.txt'"
+                    ): str,
+                    Literal(
+                        "text",
+                        description="Text to be prepended to the Google Doc."
+                    ): str
+                }
+            ),
+        }
+    )
+    def prepend_text_to_google_doc(self, params: dict) -> InfoArtifact | ErrorArtifact:
+        try:
+            file_path = params["file_path"]
+            text = params["text"]
+            service = self._build_client(self.DOCS_SCOPES, "docs", "v1")
+            drive_service = self._build_client(self.DRIVE_SCOPES, "drive", "v3")
+            document_id = self._path_to_file_id(drive_service, file_path)
+            if document_id:
+                doc = service.documents().get(documentId=document_id).execute()
+
+                if len(doc["body"]["content"]) == 1:
+                    requests = [{"insertText": {"location": {"index": 1}, "text": text}}]
+                else:
+                    start_index = doc["body"]["content"][1]["startIndex"]
+                    requests = [{"insertText": {"location": {"index": start_index}, "text": text}}]
+
+                service.documents().batchUpdate(documentId=document_id, body={"requests": requests}).execute()
+                return InfoArtifact("text prepended successfully")
+            else:
+                return ErrorArtifact(f"error prepending to google doc, file not found for path {file_path}")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"error prepending text to Google Doc with path {file_path}: {e}")
+
+    @activity(
+        config={
+            "description": "Can be used to create a new Google Doc and optionally save content to it.",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_path",
+                        description="Name of the file to be created, which will be used to save content in."
+                    ): str,
+                    Optional(
+                        "content",
+                        default=None,
+                        description="Optional content to be saved in Google Doc."
+                    ): str,
+                    Optional(
+                        "folder_path",
+                        default=DEFAULT_FOLDER_PATH,
+                        description="Path of the folder where the document will be created."
+                    ): str,
+                }
+            ),
+        }
+    )
+    def save_content_to_google_doc(self, params: dict) -> ErrorArtifact | InfoArtifact:
```

Comment:
I think this activity name should reflect that this method creates. Maybe just `create_google_doc`?

Reply:
Vasily said to leave it as `save_content_to_google_doc`, since the description already mentions the dual functionally of the function.